### PR TITLE
Copy to clipboard: move to CDK instead of current API

### PR DESCRIPTION
**Pull request by rowa-audil**
Moves copy to clipboard component to Angular's CDK.

Closes. #77

### CODE REVIEW

File: libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts [hunk 1]
```diff
@@ -53,7 +54,10 @@ const DT_COPY_TO_CLIPBOARD_SUCCESSFUL = 'dt-copy-to-clipboard-successful';
   encapsulation: ViewEncapsulation.Emulated,
 })
 export class DtCopyToClipboard implements AfterContentInit, OnDestroy {
-  constructor(private _cd: ChangeDetectorRef) {}
+  constructor(
+    private _cd: ChangeDetectorRef,
```
Comment (tomheller) [inline libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts]:
```suggestion
    private _changeDetectorRef: ChangeDetectorRef,
```

I know that you did not change this one, but as long as we are touching this file, I feel like we could align this one to our coding guidelines.

File: libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts [hunk 2]
```diff
@@ -89,26 +93,25 @@ export class DtCopyToClipboard implements AfterContentInit, OnDestroy {
...
-        );
-        this._copyButton.nativeElement.focus();
-      }
+      this._input.nativeElement.textContent !== null
+        ? this.cdkClipboard.copy(this._input.nativeElement.textContent)
+        : this.copyFailed.emit();
+    }
+  }
+
+  copiedToClipboard(): void {
```
Comment (tomheller) [inline libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts]: does this one need to be public?

File: libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts [hunk 3]
```diff
@@ -89,26 +93,25 @@ export class DtCopyToClipboard implements AfterContentInit, OnDestroy {
   /** Copies the provided content to the clipboard. */
   copyToClipboard(): void {
     if (this._input) {
```
Comment (lara-aigmueller) [inline libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts]: Should we also emit the `copyFailed` event when there is no `_input`, i.e. in the else-branch?

Comment (rowa-audil) [reply libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts]: Yes, thanks for pointing it out.

File: libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts (hunk 2 above)
Comment (rowa-audil) [reply libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts]: Its called by the CDK when it finished copying, therefor I'll annotate it with @internal in the new commit.

**Comment by tomheller** [comment]
Can you please squash the two commits into one?

**Comment by rowa-audil** [comment]
Squashed.

**Comment by lukasholzer** [comment]
please rebase to get the security fixes

File: libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts [hunk 4]
```diff
@@ -88,27 +92,27 @@ export class DtCopyToClipboard implements AfterContentInit, OnDestroy {
...
-      this._showIcon = true;
-      _addCssClass(this._input.nativeElement, DT_COPY_TO_CLIPBOARD_SUCCESSFUL);
-      if (this._copyButton) {
-        _addCssClass(
-          this._copyButton.nativeElement,
-          DT_COPY_TO_CLIPBOARD_SUCCESSFUL,
-        );
-        this._copyButton.nativeElement.focus();
-      }
+    if (this._input && this._input.nativeElement.textContent !== null) {
```
[the remark this answers is not in the document]
Comment (tomheller) [reply libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts]: I would consider this a NIT, to be honest.

**Comment by ffriedl89** [comment]
Is this really a feature? I'd argue this is more a refactor since the consumer does not care if we use the copy to clipboard internally or do it by ourselves. @dynatrace-oss/barista  What do you think? 

I'd suggest rewording the commit to refactor instead of feat and merge it afterwards
